firefox-ios/Client/Frontend/TrackingProtection/BlockedTrackersTableController.swift

```diff
@@ -0,0 +1,385 @@
+}
+
+// MARK: BlockedTrackersTableModel
+struct BlockedTrackersTableModel {
+    let topLevelDomain: String
+    let title: String
+    let URL: String
+    let contentBlockerStats: TPPageStats?
+    let connectionSecure: Bool
+
+    func getItems() -> [BlockedTrackerItem] {
+        let crossSiteCount = String(contentBlockerStats?.getTrackersBlockedForCategory(.advertising) ?? 0)
+        let fingerprintersCount = String(contentBlockerStats?.getTrackersBlockedForCategory(.fingerprinting) ?? 0)
+        let socialMediaCount = String(contentBlockerStats?.getTrackersBlockedForCategory(.social) ?? 0)
+        let trackingContentCount = String(contentBlockerStats?.getTrackersBlockedForCategory(.analytics) ?? 0)
+
+        let crossSiteText = String(format: .Menu.EnhancedTrackingProtection.crossSiteTrackersBlockedLabel,
+                                   crossSiteCount)
+        let fingerprintersText = String(format: .Menu.EnhancedTrackingProtection.fingerprinterBlockedLabel,
+                                        fingerprintersCount)
+        let socialMediaText = String(format: .Menu.EnhancedTrackingProtection.socialMediaTrackersBlockedLabel,
+                                     socialMediaCount)
+        let trackingContentText = String(format: .Menu.EnhancedTrackingProtection.analyticsTrackersBlockedLabel,
+                                         trackingContentCount)
+
+        let crossSiteImage = UIImage(
+            imageLiteralResourceName: ImageIdentifiers.TrackingProtection.crossSiteTrackers
+        ).withRenderingMode(.alwaysTemplate)
+        let fingerprintersImage = UIImage(
+            imageLiteralResourceName: ImageIdentifiers.TrackingProtection.fingerprintersTrackers
+        ).withRenderingMode(.alwaysTemplate)
+        let socialMediaImage = UIImage(
+            imageLiteralResourceName: ImageIdentifiers.TrackingProtection.socialMediaTrackers
+        ).withRenderingMode(.alwaysTemplate)
+        let trackingContentImage = UIImage(
+            imageLiteralResourceName: ImageIdentifiers.TrackingProtection.analyticsTrackersImage
+        ).withRenderingMode(.alwaysTemplate)
+
+        return [
+            BlockedTrackerItem(
+                title: crossSiteText,
+                image: crossSiteImage
+            ),
+            BlockedTrackerItem(
+                title: fingerprintersText,
+                image: fingerprintersImage
+            ),
+            BlockedTrackerItem(
+                title: trackingContentText,
+                image: trackingContentImage
+            ),
+            BlockedTrackerItem(
+                title: socialMediaText,
+                image: socialMediaImage
+            )
+        ]
+    }
+}
+
+// MARK: BlockedTrackersTableViewController
+class BlockedTrackersTableViewController: UIViewController,
+                                          Themeable,
+                                          UITableViewDelegate {
+    private lazy var blockedTrackersTableView: UITableView = {
```

Comment:
to avoid this VC from getting too big, what do you think about separating this out into its own class + file like the header view